common/src/main/java/org/jboss/hal/testsuite/testlistener/TestListener.java

```diff
@@ -0,0 +1,64 @@
+package org.jboss.hal.testsuite.testlistener;
+
+import org.apache.commons.io.FileUtils;
+import org.jboss.hal.testsuite.creaper.ManagementClientProvider;
+import org.jboss.hal.testsuite.util.PropUtils;
+import org.junit.runner.Description;
+import org.junit.runner.notification.RunListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+
+import java.io.File;
+import java.io.IOException;
+
+public class TestListener extends RunListener {
+
+    private static final String XML_SNAPSHOT_DIRECTORY = PropUtils.get("snapshots.target.directory.name");
+
+    private static final Logger log = LoggerFactory.getLogger(TestListener.class);
+
+    private Class<?> testClazz;
+
+    @Override
+    public void testStarted(Description description) throws Exception {
+        if (this.testClazz == null) {
+            this.testClazz = description.getTestClass();
+        } else if (!this.testClazz.equals(description.getTestClass())) { //started new testcase
+            finishedTestClassRun(this.testClazz); //save configuration for previous test class
+            this.testClazz = description.getTestClass();
+            log.debug("Started running new test class: '{}'!", this.testClazz.getCanonicalName());
+        }
+    }
+
+    private void finishedTestClassRun(Class<?> testClazz) throws IOException {
+        File directory = new File(FileUtils.getTempDirectoryPath()
+                .concat(File.separator)
+                .concat(XML_SNAPSHOT_DIRECTORY)
+                .concat(File.separator)
+                .concat(testClazz.getCanonicalName()));
+        if (!directory.exists()) {
+            if (!directory.mkdirs()) {
+                throw new IOException("Unable to create path '" + directory.toString() + "'.");
+            }
+        }
+        File copyTo = File.createTempFile(PropUtils.get("snapshots.target.file.prefix"),
+                PropUtils.get("snapshots.target.file.suffix"), directory);
```

Comment:
Maybe putting the time info already to directory name, to easily find out order of testcases they were run in.